core/src/main/java/org/apache/calcite/plan/VisitorDataContext.java

```diff
@@ -75,12 +81,27 @@ public static DataContext of(RelNode targetRel, LogicalFilter queryRel) {
   }
 
   public static DataContext of(RelDataType rowType, RexNode rex) {
+    return of(rowType, rex, null, null);
+  }
```

Comment:
useless constructor

Reply:
This has come from the previous definition. I think useless is somewhat subjective expression. This method is a public static method in a public class. I personally don't make this function call, but someone might be already using it. I didn't want to abruptly remove a public method. Also, giving nulls to builder and executor would make it possible to bypass the executor so that the output you get from it would be exactly the same as before. So, I think this can be useful especially for those who think this change is not necessary :) That being said, I am okay with removing existing 'of' methods if the community is okay with it.

Also, as far as I know, this can never be a constructor.